contracts/ERC721M.sol

```diff
@@ -371,4 +404,32 @@ contract ERC721M is IERC721M, ERC721AQueryable, Ownable {
             )
         ) revert InvalidCosignSignature();
     }
+
+    function getActiveStageFromTimestamp(uint256 timestamp)
+        public
+        view
+        returns (uint256)
+    {
+        for (uint256 i = 0; i < _mintStages.length; i++) {
+            if (
+                timestamp >= _mintStages[i].startTimeUnixSeconds &&
```

Comment:
nit: since we check on write path, read path only needs to check `<`

Reply:
i actually think we need the `>=` check in case we are in between stages, or stage hasn't started yet